packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx

```diff
@@ -226,13 +226,24 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
   };
 
   const handleTitleClick = e => {
-    if (hasSubtree && e.target === e.currentTarget) {
+    if (selectable) {
+      if (hasSubtree) {
+        // need to know if click happened on selection indicator or title itself
+        if (e?.target?.className?.includes(treeTitleSlotClassNames.indicator)) {
+          toggleItemSelect(e, id);
+        } else {
+          toggleItemActive(e, id);
+        }
+      } else {
+        toggleItemSelect(e, id);
+      }
+    } else if (hasSubtree) {
       toggleItemActive(e, id);
```

Comment:
I dunno I find it more readable without that many if/else but maybe that is just me 🤷🏻‍♂️ ?

```suggestion
        // need to know if click happened on selection indicator or title itself
        if (e?.target?.className?.includes(treeTitleSlotClassNames.indicator || !hasSubtree && selectable)) {
          toggleItemSelect(e, id);
        } else {
          toggleItemActive(e, id);
        }
```

Reply:
The updated logic is not exactly the same.
The idea is: when tree item is not selectable, and has sub tree, `handleTitleClick` always expand/collapse.
When it is selectable, clicking on item itself will expand/collapse, clicking on the selection indicator box will select/unselect.
Any ideas to make a readable logic out of it?

Reply:
If `Tree` is not selectable and has subtree it would go inside second `if` so it would expand/collapse or I am missing something?

Reply:
Another way to see it could be: 

```suggestion
    // need to know if click happened on selection indicator or title itself
        const shouldSelect =
      selectable && (e?.target?.className?.includes(treeTitleSlotClassNames.indicator) || !hasSubtree);

    if (shouldSelect) {
      toggleItemSelect(e, id);
    }

    if (!shouldSelect) {
      toggleItemActive(e, id);
    }

```

Reply:
I set `House of Lannister` to not selectable and tested in your branch, both logics seems fine: 

![PbvUuCHTdM](https://user-images.githubusercontent.com/8545105/101339512-4d01a700-385d-11eb-9c6d-5750161cff46.gif)

Reply:
Love the new change❤️ Going to commit it. But don't approve it for now because I plan to add UT tomorrow as we don't have it for selectable tree.